### PR TITLE
#85 Override executed script

### LINKED ISSUE #85: Checksum file changed but filename did not
When a new checksum is found, check the name of the file. If it has the same name than a script already executed, update the checksum in the database.
Maybe also do not do not execute it.

### PR DESCRIPTION
**Pull request by driccio**
Override executed script 

Resolves/Fixes/Closes #85

### CODE REVIEW

**Review by Lysoun — CHANGES_REQUESTED** [review]
Since this can't be in a review comment, I'm telling here: the action of ```DatamaintainConfig``` should be added in the ```log``` function of the class.

I don't see anything regarding the new option for the configuration to enable Datamaintain to choose to override executed scripts, is that normal?

File: modules/core/src/main/kotlin/datamaintain/core/script/Script.kt [hunk 1]
```diff
@@ -42,6 +42,7 @@ data class ExecutedScript @JvmOverloads constructor(
                         script.checksum,
                         script.identifier,
                         execution.executionStatus,
+                        script.action,
                         executionDurationInMillis,
                         execution.executionOutput
                 )
```
[the remark this answers is not in the document]
Comment (driccio) [reply modules/core/src/main/kotlin/datamaintain/core/script/Script.kt:48]: I agree with you. But I will not manage that in my PR.

File: modules/core/src/main/kotlin/datamaintain/core/step/executor/Executor.kt [hunk 2]
```diff
@@ -16,43 +15,70 @@ class Executor(private val context: Context) {
...
-                    var execution = Execution(ExecutionStatus.SHOULD_BE_EXECUTED)
-                    val executionDurationInMillis = measureTimeMillis {
-                        execution = context.dbDriver.executeScript(script)
-                    }
-                    ExecutedScript.build(script, execution, executionDurationInMillis)
-                }
-                ExecutionMode.FORCE_MARK_AS_EXECUTED -> ExecutedScript.forceMarkAsExecuted(script)
-                ExecutionMode.DRY -> ExecutedScript.shouldBeExecuted(script)
+                ExecutionMode.NORMAL -> doAction(script)
+                else -> simulateAction(script)
```
[the remark this answers is not in the document]
Comment (driccio) [reply modules/core/src/main/kotlin/datamaintain/core/step/executor/Executor.kt:22]: You're absolutely right.

File: modules/core/src/main/kotlin/datamaintain/core/script/FileScript.kt [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
 import java.math.BigInteger
 import java.nio.file.Path
 import java.security.MessageDigest
 
 class FileScript @JvmOverloads constructor(
         val path: Path,
         identifierRegex: Regex,
-        override val tags: Set<Tag> = setOf()
+        override val tags: Set<Tag> = setOf(),
+        override var action: ScriptAction = ScriptAction.RUN
```
[the remark this answers is not in the document]
Comment (driccio) [reply modules/core/src/main/kotlin/datamaintain/core/script/FileScript.kt:11]: I agree with you but it extends ScriptWithContent that needs this information.

File: modules/cli/src/main/kotlin/datamaintain/cli/App.kt [hunk 4]
```diff
@@ -71,6 +72,8 @@ class UpdateDb(val runner: (DatamaintainConfig) -> Unit = ::defaultUpdateDbRunne
 
     private val executionMode by option(help = "execution mode").choice(ExecutionMode.values().map { it.name }.map { it to it }.toMap())
 
+    private val action by option(help = "script action").choice(ScriptAction.values().map { it.name }.map { it to it }.toMap())
+
```
[the remark this answers is not in the document]
Comment (Lysoun) [reply modules/cli/src/main/kotlin/datamaintain/cli/App.kt:76]: An issue was created to deal with that: #121 
I am resolving the conversation

**Review by nroulon — APPROVED** [review]

**Pull request merged**